crates/bevy_ui/src/ui_node.rs

```diff
@@ -575,6 +575,38 @@ pub struct Style {
     ///
     /// <https://developer.mozilla.org/en-US/docs/Web/CSS/grid-column>
     pub grid_column: GridPlacement,
+
+    /// Used to add rounded corners to a UI node. You can set a UI node to have uniformly rounded corners
+    /// or specify different radii for each corner. If a given radius exceeds half the length of the smallest dimension between the node's height or width,
+    /// the radius will calculated as half the smallest dimension.
+    ///
+    /// Elliptical nodes are not supported yet. Percentage values are based on the node's smallest dimension, either width or height.
+    ///
+    /// ```
+    /// # Example
+    /// ```
+    /// # use bevy_ui::{Style, UiRect, Val};
+    /// let style = Style {
+    ///     // Set a uniform border radius of 10 logical pixels
+    ///     border_radius: UiBorderRadius::all(Val::Px(10.)),
+    ///     ..Default::default()
+    /// };
+    /// let style = Style {
+    ///     border_radius: UiBorderRadius {
+    ///         // The top left corner will be rounded with a radius of 10 logical pixels.
+    ///         top_left: Val::Px(10.),
+    ///         // Percentage values are based on the node's smallest dimension, either width or height.
+    ///         top_right: Val::Percent(20.),
+    ///         // Viewport coordinates can also be used.
+    ///         bottom_left: Val::Vw(10.),
+    ///         // The bottom right corner will be unrounded.
+    ///         ..Default::default()
+    ///     },
+    /// };
+    /// ```
+    ///
+    /// <https://developer.mozilla.org/en-US/docs/Web/CSS/border-radius>
+    pub border_radius: UiBorderRadius,
```

Comment:
I initially had `UiBorderRadius` as a separate component.
But I think from a user POV (with no idea we use Taffy etc) it makes sense that  `border` and `border_radius` should belong together. It's only because of how the implementation and change detection work that we are grouping together some properties in `Style` and making properties components really.

Also `UiBorderRadius` isn't just visual. I haven't merged this part yet here but the focus system takes into account rounded nodes, so that when you are hovering over a node with a rounded boundary when you are inside the rounded boundary, not the rectangular bounds.

Reply:
I  would prefer a very granular approach with a lot of independent components too though.

It's just that while we have a monolithic style component, it feels confused not to have everything join the monolith.

Reply:
`BorderColor` is a distinct component like `BackgroundColor`. If colors were stored in `Style` then changing a color would trigger a layout update because of change detection.

Reply:
> BorderColor is a distinct component like BackgroundColor. If colors were stored in Style then changing a color would trigger a layout update because of change detection.

This also applies to`BorderRadius`, no? Currently changing a border radius will trigger a layout (which it shouldn't). Admittedly, one is less likely to want to change a border radius frequently than a border color.

Reply:
This is another example of how renaming `Style` to be more specific would clear up a lot of confusion: #10497 - if `Style` were renamed to `LayoutConstraints` then we wouldn't be having this discussion.